backup-stores/s3/src/test/java/io/camunda/zeebe/backup/s3/BackupUploadIT.java

```diff
@@ -0,0 +1,171 @@
+    }
+  }
+
+  @AfterEach
+  void tearDown() {
+    store.closeAsync();
+  }
+
+  public void setConfigParallelConnectionsAndTimeout(
+      final int parallelConnections, final Duration timeout) {
+    config =
+        new Builder()
+            .withBucketName(BUCKET_NAME)
+            .withBasePath(RandomStringUtils.randomAlphabetic(10).toLowerCase())
+            .withEndpoint(S3.getEndpointOverride(Service.S3).toString())
+            .withRegion(S3.getRegion())
+            .withCredentials(S3.getAccessKey(), S3.getSecretKey())
+            .withApiCallTimeout(null)
+            .forcePathStyleAccess(false)
+            .withCompressionAlgorithm(null)
+            .withConnectionAcquisitionTimeout(timeout)
+            .withParallelUploadsLimit(parallelConnections)
+            .build();
+    client = S3BackupStore.buildClient(config);
+    store = new S3BackupStore(config, client);
+  }
+
+  @Test
+  void shouldSaveBackupWithManyFiles() throws IOException {
+    // given
+    // Default values for the configuration
+    setConfigParallelConnectionsAndTimeout(50, Duration.ofSeconds(10));
+    final CompletableFuture<Void> saveFuture = getStore().save(largeNumberOfSegmentsBackup(4_000));
+
+    // then
+    Assertions.assertThat(saveFuture).succeedsWithin(Duration.ofSeconds(60));
+  }
+
+  @Test
+  void shouldNotTimeoutForAcquisitionOfConnection() throws IOException {
+    // given
+    // Even with just one connection, and low timeout limit, the second upload should not start
+    // until a connection is available, and therefore should not throw AcquisitionConnectionTimeout
+    setConfigParallelConnectionsAndTimeout(1, Duration.ofMillis(50));
+    final CompletableFuture<Void> saveFuture = getStore().save(configurableLargeBackup(500_000));
+
+    // then
+    Assertions.assertThat(saveFuture).succeedsWithin(Duration.ofSeconds(60));
+  }
+
+  Backup largeNumberOfSegmentsBackup(final int numberOfSegments) throws IOException {
+    final var tempDir = Files.createTempDirectory("backup");
+    Files.createDirectory(tempDir.resolve("segments/"));
+    Files.createDirectory(tempDir.resolve("snapshot/"));
+    final Map<String, Path> largeNumberOfSegments = new HashMap<>();
+    final var s1 = Files.createFile(tempDir.resolve("snapshot/snapshot-file-1"));
+    final var s2 = Files.createFile(tempDir.resolve("snapshot/snapshot-file-2"));
+
+    for (int i = 0; i < numberOfSegments; i++) {
+      largeNumberOfSegments.put(
+          "segment-file-%d".formatted(i),
+          Files.createFile(tempDir.resolve("segments/segment-file-%d".formatted(i))));
+    }
+
+    return new BackupImpl(
+        new BackupIdentifierImpl(1, 2, 3),
+        new BackupDescriptorImpl(Optional.of("test-snapshot-id"), 4, 5, "test"),
+        new NamedFileSetImpl(largeNumberOfSegments),
+        new NamedFileSetImpl(Map.of("snapshot-file-1", s1, "snapshot-file-2", s2)));
+  }
+
+  Backup configurableLargeBackup(final int sizeOfFileInBytes) throws IOException {
```

Comment:
:wrench: give the two backup-creating methods a similar name. I'd suggest `backupWithManyFiles` and `backupWithLargeFiles`